core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -142,15 +145,21 @@ memories_deinstantiate(WASMModuleInstance *module_inst,
                 }
                 if (memories[i]->memory_data) {
 #ifndef OS_ENABLE_HW_BOUND_CHECK
-                    wasm_runtime_free(memories[i]->memory_data);
-#else
-#ifdef BH_PLATFORM_WINDOWS
-                    os_mem_decommit(memories[i]->memory_data,
-                                    memories[i]->num_bytes_per_page
-                                        * memories[i]->cur_page_count);
+#ifdef WASM_LINEAR_MEMORY_MMAP
+                    if (shared_memory_is_shared(memories[i])) {
+                        map_size = (uint64)memories[i]->num_bytes_per_page
+                                   * memories[i]->max_page_count;
+                        wasm_munmap_linear_memory(memories[i]->memory_data,
+                                                  map_size, map_size);
+                    }
+                    else
 #endif
-                    os_munmap((uint8 *)memories[i]->memory_data,
-                              8 * (uint64)BH_GB);
+                        wasm_runtime_free(memories[i]->memory_data);
+#else
+                    map_size = memories[i]->num_bytes_per_page
```

Comment:
Had better be `(uint64)memories[i]->num_bytes_per_page * memories[i]->cur_page_count` to avoid possible integer overflow?